astropy/visualization/wcsaxes/helpers.py

```diff
@@ -24,6 +28,56 @@
 }
 
 
+def add_compass(
+    ax,
+    north_angle,
```

Comment:
Rather than passing in ``north_angle`` I think that we should just compute the angle from ``ax.wcs`` - see also my comment about removing the ``north_pole_angle`` function.